### PR TITLE
Push/documentation

### PR DESCRIPTION
**Pull request by marto83**
I've added a main document describing how to setup push notifications for your app based on [the documentation](https://github.com/ably/ably-flutter/blob/main/PushNotifications.md) @ben-xD wrote for [Flutter](https://github.com/ably/ably-flutter).

Remaining:

- Notification Publisher Readme
- DotNetPush sample app Readme

The PR will become smaller once the iOS branch gets merged.

### CODE REVIEW

File: Push README.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+### Receiving Messages
+
+For examples of handling incoming messages and dealing with notifications, see [push_notification_handlers](example/lib/push_notifications/push_notification_handlers.dart) in the example app.
+
+#### Notification Message / Alert Push Notification
+
+**Android**: If the app is in the background / terminated, you cannot configure / disable notification messages as they are automatically shown to the user by Firebase Messaging Android SDK. To create notifications which launch the application to a certain page (notifications which contain deep links or app links), or notifications which contain buttons / actions, images, and inline replies, you should send a data message and create a notification when the message is received. 
+
+**iOS**: If the app is in the background / terminated, the ably sdk doesn't provide the functionality to configure / extend alert notifications on iOS, and these will automatically be shown to the user.
```
Comment (tomkirbygreen) [inline Push README.md:197]: "ably" => "Ably".

File: examples/NotificationsPublisher/Readme.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Running the app
+
+You can start the app using `dotnet run` 
+
+## Functionality 
+
+When you run the app for the first time you will be asked to enter an Ably `key` which will be saved in a file called `key.secret` in the current directory. 
+Once the key is entered and stored the app can: 
+- Connect to ably
```
Comment (tomkirbygreen) [inline examples/NotificationsPublisher/Readme.md:16]: "ably" => "Ably".

**Review by tomkirbygreen — CHANGES_REQUESTED** [review]

File: Push README.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+### Android
+
+- Open the DotnetPush solution which can be found in the `examples` folder in the repository, make sure to change the application Id in the Manifest file.
+- Create a Firebase project, and in the Project settings, add an android app with your unique application ID. Follow the steps provided on the setup process, or the following:
+    - You can leave `Debug signing certificate SHA-1` empty.
+    - Download the generated `google-services.json` file
+    - Place `google-services.json` in `example/android/app/`. We have `gitignore`d this file since it is associated with our Firebase project, but it is [not sensitive](https://stackoverflow.com/questions/37358340/should-i-add-the-google-services-json-from-firebase-to-my-repository), so you can commit it to share it with other developers/colleagues.
+- Provide ably with the FCM server key: In your [firebase project settings](https://knowledge.ably.com/where-can-i-find-my-google/firebase-cloud-messaging-api-key), create or use an existing cloud messaging server key, and enter it in your Ably app's dashboard (App > Notifications tab > Push Notifications Setup > Setup Push Notifications).
```
[the remark this answers is not in the document]
Comment (marto83) [reply Push README.md]: Updated

File: Push README.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
+# Push Notifications
+
+Push Notifications allow you to reach users who have your application in the foreground, background and terminated, including when your application is not connected to Ably. Push notifications allow you to run code and show alerts to the user. On iOS, Ably connects to [APNs](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/APNSOverview.html) to send messages to devices. On Android, Ably connects to [Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging/) to send messages to devices. As both services do not guarantee message delivery and may even throttle messages to specific devices based on battery level, message frequency, and other criteria, messages may arrive much later than sent or ignored.
```
[the remark this answers is not in the document]
Comment (marto83) [reply Push README.md]: I've updated the first paragraph

File: Push README.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+## Known Limitations
+ 
+- Requires initialising AblyRealtime: Currently activating push notifications for a device requires an instance of AblyRealtime which gets instantiated when the application starts up. Please, submit an issue if you do not require a realtime instance and would like to use the xamarin push notifications without using an AblyRealtime connection. For the time being when calling `(AndroidMobileDevice|AppleMobileDevice).Initialise` please set `ClientOptions.AutoConnect` to `false` which will keep the AblyRealtime instance from using any resources.
```
[the remark this answers is not in the document]
Comment (marto83) [reply Push README.md]: Done.

File: Push README.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+        - Check `remote notifications`.
+- The above is also described in [our native ios tutorial](https://ably.com/tutorials/ios-push-notifications#title).
+
+## Usage
+
+### Summary
+
+Devices need to be [activated](#device-activation) with Ably once. Once activated, you can use their device ID, client ID or push token (APNs device token/ FCM registration token) to push messages to them using the Ably dashboard or a [Push Admin](https://ably.com/documentation/general/push/admin) (SDKs which provide push admin functionality, such as the .net sdk, [Ably-java](https://github.com/ably/ably-java), [Ably-js](https://github.com/ably/ably-js), etc.). However, to send push notifications through Ably channels, devices need to [subscribe to a channel for push notifications](#subscribing-to-channels-for-push-notifications). Once subscribed, messages on that channel with a [push payload](#sending-messages) will be sent to devices which are subscribed to that channel.
+
+The DotNetPush app contains an example of how to use the Push Notification functionality. (TODO: Add links to github once PRs have been approved)
```
[the remark this answers is not in the document]
Comment (marto83) [reply Push README.md]: I've added links to the DotnetPush cs files

File: Push README.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+### Device activation
+
+- The first step is to initialise the specific mobile device. You can achieve this by using either `AppleMobileDevice.Initialise` or `AndroidModileDevice.Initialise`. To initialise you need to pass a valid Ably `ClientOptions` which will be used to initialise the ably realtime client and an instance of `PushCallbacks` which you can use to recieve notification when the device is `Activated`, `Deactivated` or `SyncRegistrationFailed`.
```
[the remark this answers is not in the document]
Comment (marto83) [reply Push README.md]: Done

File: Push README.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+#### Notification Message / Alert Push Notification
+
+Shows a notification to the user immediately when it is received by their device.
+
+**Android**: This is known as a [notification message](https://firebase.google.com/docs/cloud-messaging/concept-options). A notification message cannot be customised or handled (e.g. run logic when a user taps the notification) - therefore, if you need to handle user taps or customize a notification, send a data message and [create a local notification](https://developer.android.com/guide/topics/ui/notifiers/notifications).
+
+**iOS**: This is known as an [alert push notification](https://developer.apple.com/documentation/usernotifications/). An alert notification can be [customised](https://developer.apple.com/documentation/usernotificationsui/customizing_the_appearance_of_notifications).
+
+This is how you construct a message with a push notification. Normal subscribers will only receive the message where devices subscribed to that channel will receive the Push notification.
+```csharp
```
[the remark this answers is not in the document]
Comment (marto83) [reply Push README.md:130]: Done

File: Push README.md (hunk 1 above)
Comment (marto83) [reply Push README.md]: Done

File: examples/NotificationsPublisher/Readme.md (hunk 2 above)
Comment (marto83) [reply examples/NotificationsPublisher/Readme.md]: Done

**Review by tomkirbygreen — APPROVED** [review]

**Pull request merged**